plugins/scaffolder/src/components/TemplatePage/TemplatePage.tsx

```diff
@@ -27,28 +27,26 @@ import { catalogApiRef } from '@backstage/plugin-catalog';
 import { LinearProgress } from '@material-ui/core';
 import { IChangeEvent } from '@rjsf/core';
 import React, { useState } from 'react';
+import { Navigate } from 'react-router';
 import { useParams } from 'react-router-dom';
-import useStaleWhileRevalidate from 'swr';
+import { useAsync } from 'react-use';
 import { scaffolderApiRef } from '../../api';
-import { JobStatusModal } from '../JobStatusModal';
+import { rootRoute } from '../../routes';
 import { Job } from '../../types';
+import { JobStatusModal } from '../JobStatusModal';
 import { MultistepJsonForm } from '../MultistepJsonForm';
-import { Navigate } from 'react-router';
-import { rootRoute } from '../../routes';
 
 const useTemplate = (
   templateName: string,
   catalogApi: typeof catalogApiRef.T,
 ) => {
-  const { data, error } = useStaleWhileRevalidate(
-    `templates/${templateName}`,
-    async () =>
-      catalogApi.getEntities({
-        kind: 'Template',
-        'metadata.name': templateName,
-      }) as Promise<TemplateEntityV1alpha1[]>,
-  );
-  return { template: data?.[0], loading: !error && !data, error };
+  const { value, loading, error } = useAsync(async () => {
```

Comment:
@benjdlambert Note that I changed from `swr` to `useAsync` here. The tests were horribly misbehaving and had cross dependencies because of weird swr issues - and after fighting for a while with it, I just changed the code here instead. It didn't feel like it actually had any particular added value over useAsync at this point, so the simplicity was just a bonus.

Reply:
I think it was then you click the tabs in the Catalog table it jumps around and flickers because there's no data IIRC.

Reply:
Circling back to this after I saw it was merged, realised that the usage of `swr` has grown exponentially, and that this usage wasn't in the place I thought it was.

I'd be interested to work out why the tests were misbehaving - but I guess I wonder if it's actually worth investigating wrapping these clients that we will create with the ability to respect the `cache-control` headers and have their own handling instead of `swr`?

Reply:
Cache headers is just one source of info about when it's appropriate to update, I think. Yeah it's probably worth looking into in one form or the other either way. Bear in mind as well, btw, that probably (hopefully?) a lot of this will be driven through graphql eventually, which also has its own mind when it comes to caching.

Reply:
Yeah you're right.

The `GraphQL` data fetching layer however would ideally just lift the implementation of caching from the frontend and move that up the stack a little bit. It does also pose another factor too though, considering frameworks like `apollo` in the frontend, which has the mentality of having to explicitly say `i would like to have the latest from the server please` instead of relying on the smart internal `apollo` caching where it keeps a copy of the graph client side to stop network requests where it should already have the data.

I guess what i'm trying to say, is that we're going to have two seperate layers of caching to think about soon, and maybe the `swr` wasn't the best solution for this anyway.